src/api/operator/numpy/np_fill_diagonal_op.cc

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file np_fill_diagonal_op.cc
+ * \brief Implementation of the API of functions in src/operator/numpy/np_fill_diagonal.cc */
+#include <mxnet/api_registry.h>
+#include "../utils.h"
+#include "../../../operator/numpy/np_fill_diagonal_op-inl.h"
+
+namespace mxnet {
+
+MXNET_REGISTER_API("_npi.fill_diagonal")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_fill_diagonal");
+  nnvm::NodeAttrs attrs;
+
+  op::NumpyFillDiagonalParam param;
+  int num_inputs = 1;
+  NDArray* inputs[] = {args[0].operator mxnet::NDArray*()};
+
+  param.val = Tuple<double>(1, args[1].operator double());
```

Comment:
tuple of float is not supported for now. We may wait for #17866 .

Reply:
#17866 merged

Reply:
@BenjaminCHEN2016 #17866 has been merged. Let's use `Obj2Tuple` like https://github.com/apache/incubator-mxnet/pull/17866/files#diff-4c8cf94cd8bf4368b07f43c165402239R49.